src/aelin-pool.ts

```diff
@@ -1,369 +1,363 @@
+    poolCreatedEntity.totalSupply = poolCreatedEntity.totalSupply.plus(event.params.value)
+  }
+  if (event.params.to.toHex() == ZERO_ADDRESS.toHex()) {
+    poolCreatedEntity.totalSupply = poolCreatedEntity.totalSupply.minus(event.params.value)
+  }
+  poolCreatedEntity.save()
+}
+
+export function handleCreateDeal(event: CreateDealEvent): void {
+  createEntity(Entity.DealCreated, event)
+  createEntity(Entity.DealSponsored, event)
+  createEntity(Entity.Deal, event)
+
+  /**
+   * Update PoolCreated entity
+   */
+  let poolCreatedEntity = getPoolCreated(event.address.toHex())
+  if (poolCreatedEntity != null) {
+    poolCreatedEntity.poolStatus = PoolStatus.FundingDeal
+    poolCreatedEntity.dealAddress = event.params.dealContract
+    poolCreatedEntity.save()
+  }
+
+  /**
+   * TotalDealsBySponsor entity
+   */
+  let totalDealsBySponsorEntity = getTotalDealsBySponsor(event.params.sponsor.toHexString())
+  if (totalDealsBySponsorEntity == null) {
+    createEntity(Entity.TotalDealsBySponsor, event)
+    totalDealsBySponsorEntity = getTotalDealsBySponsor(event.params.sponsor.toHexString())
+  }
+  if (totalDealsBySponsorEntity != null) {
+    totalDealsBySponsorEntity.count++
+    totalDealsBySponsorEntity.save()
+  }
+
+  removeNotificationsForEvent(event)
+}
+
+export function handleDealDetail(event: DealDetailEvent): void {
+  createEntity(Entity.DealDetail, event)
+  createEntity(Entity.Deal, event)
+
+  /**
+   * Update PoolCreated entity
+   */
+  let dealCreatedEntity = getDealCreated(event.params.dealContract.toHex())
+  if (dealCreatedEntity != null) {
+    let poolCreatedEntity = getPoolCreated(dealCreatedEntity.poolAddress.toHex())
+    if (poolCreatedEntity != null) {
+      poolCreatedEntity.deal = event.params.dealContract.toHex()
+      poolCreatedEntity.holder = event.params.holder
+      poolCreatedEntity.save()
+    }
+  }
+
+  // use templates to create a new deal to track events
+  AelinDeal.create(event.params.dealContract)
+  createNotificationsForEvent(event)
+}
+
+// @TODO add block or timestamp to these events????
+// @TODO look into what needs to be indexed
+export function handlePurchasePoolToken(event: PurchasePoolTokenEvent): void {
+  createEntity(Entity.PurchasePoolToken, event)
+  createEntity(Entity.Deposit, event)
+
+  /**
+   * Update PoolCreated entity
+   */
+  let poolCreatedEntity = getPoolCreated(event.address.toHex())
+  if (poolCreatedEntity != null) {
+    poolCreatedEntity.contributions = poolCreatedEntity.contributions.plus(
+      event.params.purchaseTokenAmount,
+    )
+
+    const totalAddressesInvested = poolCreatedEntity.totalAddressesInvested
+    const hasInvested = totalAddressesInvested.includes(event.params.purchaser.toHex())
+    if (!hasInvested) {
+      poolCreatedEntity.totalUsersInvested++
+      totalAddressesInvested.push(event.params.purchaser.toHex())
+      poolCreatedEntity.totalAddressesInvested = totalAddressesInvested
+      poolCreatedEntity.save()
```

Comment:
The `poolCreatedEntity.save()` should be outside the if statement, otherwise it will not display the correct number of funds if the same address deposits more than once